### DISCUSSION TITLE
How modify environment variables on change cwd?

### DISCUSSION THREAD

**Discussion by JMarkin** [sxyazi/yazi#938] [1. Q&A]
Hello, in my fish shell workflow after change cwd I can automatically run function like export from .env file. 

How I can do it in yazi?

**Comment by sxyazi** [comment]
Hi, I'm not very sure I understand what you mean. If you're looking to execute some custom commands when the Yazi cwd changes, you can subscribe to the `cd` event using `ps.sub("cd", function() end)`. Here's an example: https://github.com/sxyazi/yazi/blob/cb92bb798089dd5ce2b037fb6d2b5991e2153d00/yazi-plugin/preset/plugins/zoxide.lua#L38-L47

Doc of `ps.sub()`: https://yazi-rs.github.io/docs/plugins/utils#ps.sub

Note that DDS is a new feature in the upcoming release of Yazi 0.2.5, and the latest code is currently required

**Reply by cbr9** [reply]
This doesn't seem to have any effect...

**Reply by sxyazi** [reply]
@cbr9 How did you use it exactly? Have you enabled it in your `init.lua`?

**Reply by cbr9** [reply]
Hi, yes after that I realized I had to enable in init.lua.

no worries
